test/adapter/json/collection_test.rb

```diff
@@ -85,6 +85,20 @@ def test_include_option
 
             assert_equal(expected, actual)
           end
+
+          def test_fields_with_no_associations_include_option
+            serializer = ArraySerializer.new([@first_post, @second_post])
+            adapter = ActiveModel::Serializer::Adapter::Json.new(serializer)
+            actual = adapter.serializable_hash(fields: [:id])
```

Comment:
I can fix it. But I should inform you that fields are working out of the box because there is a bug here: https://github.com/rails-api/active_model_serializers/blob/master/lib/action_controller/serialization.rb#L50-L51

and not because the adapter's constructor passes the fields option to the serializer. (the adapter's options are in instance_options whereas the serializer options are in options all Attributes methods)